psd-web/spec/features/registration_spec.rb

```diff
@@ -37,9 +37,24 @@
 
       expect_to_be_on_secondary_authentication_page
 
+      click_link "Not received a text message?"
+
+      expect_to_be_on_resend_secondary_authentication_page
+      find("details summary", text: "Change where the text message is sent").click
+      fill_in "Mobile number", with: "70123456789"
+
+      click_button "Resend security code"
+      expect_to_be_on_secondary_authentication_page
       enter_secondary_authentication_code(invitee.reload.direct_otp)
 
       expect_to_be_on_declaration_page
+      check "I agree"
+      click_button "Continue"
+      click_link "Skip introduction"
+      click_link("Your account", match: :first)
+
+      expect(page).to have_h1("Your account")
+      expect(page).to have_css("dd", text: "70123456789")
```

Comment:
We have a [useful matcher](https://github.com/UKGovernmentBEIS/beis-opss-psd/blob/master/psd-web/spec/support/matchers/capybara_matchers.rb#L7-L30) for this, which helps to assert which key in the summary table this is a value for:

```suggestion
      expect(page).to have_summary_item(key: "Mobile number", text: "70123456789")      
```